### PR TITLE
Clarification: new mean option is synonym with sum_over_batch_size in loss function base class

### PR DESCRIPTION
**Pull request by jm-willy**
Since `sum_over_batch_size` actually is `mean`, `mean` option was added as equivalent of `sum_over_batch_size` to explicitly and clearly state what `sum_over_batch_size` does, under the expected name `mean`, while keeping `sum_over_batch_size` for backwards compatibility. See also https://github.com/keras-team/keras/issues/18818, where the problem with the current name is noted.

Also, why is done like this:
```
loss = ops.sum(values)
loss /= ops.cast(
    ops.prod(ops.convert_to_tensor(ops.shape(values), dtype="int32")),
    loss.dtype,
)
```
if there's the clearer and shorter keras mean?
`loss = ops.mean(values, axis=None, keepdims=False)`

Please take it, `sum_over_batch_size` is confusing, I already implemented `mean` when I realized it was already done under a different name. 

Edit: already undid formating while keeping the typo fixes see last commit: https://github.com/keras-team/keras/pull/20352/commits/19933d94523d482ac7642913467d1a015e54b882

### CODE REVIEW

File: keras/src/losses/loss.py [hunk 1]
```diff
@@ -132,7 +133,7 @@ def reduce_values(values, reduction="sum_over_batch_size"):
     ):
         return values
     loss = ops.sum(values)
-    if reduction == "sum_over_batch_size":
+    if (reduction == "sum_over_batch_size") or (reduction == "mean"):
```
Comment (fchollet) [inline keras/src/losses/loss.py:136]: `if reduction in ("mean", "sum_over_batch_size"):`

File: keras/src/losses/loss.py [hunk 2]
```diff
@@ -177,7 +178,7 @@ def apply_mask(sample_weight, mask, dtype, reduction):
     """Applies any mask on predictions to sample weights."""
     if mask is not None:
         mask = ops.cast(mask, dtype=dtype)
-        if reduction == "sum_over_batch_size":
+        if (reduction == "sum_over_batch_size") or (reduction == "mean"):
```
Comment (fchollet) [inline keras/src/losses/loss.py:181]: Same here

**Review by fchollet — APPROVED** [review]
